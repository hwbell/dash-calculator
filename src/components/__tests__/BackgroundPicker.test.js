import { mount } from '@vue/test-utils'
import Base from '../Base.vue'

const message = 'This is a base component for us to start with!';

describe('Base', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(Base)
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  
  test('renders a button for each object in the array map', () => {
    let messageDiv = wrapper.findAll('.base-component');
    expect(messageDiv.length).toBe(1);
    // expect(messageDiv.text()).toBe('This is a base component for us to start with!')
  })

  test(`assigns the computed value on the component properly`, () => {
    let revMessage = message.split('').reverse().join('');
    expect(wrapper.vm.reversedMessage).toBe(revMessage);
  })  

  test('should change the message when the greet method is called', () => {
    wrapper.vm.greet();
    expect(wrapper.vm.message).toBe('Nice to meet you!');
  })
})