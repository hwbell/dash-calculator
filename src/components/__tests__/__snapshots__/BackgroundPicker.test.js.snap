// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders correctly 1`] = `
<div
  class="base-component"
>
  
  This is a base component for us to start with!

</div>
`;
